sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -1040,3 +1040,35 @@ def test_init():
                                max_iter=1, random_state=random_state).fit(X)
 
         assert gmm2.lower_bound_ >= gmm1.lower_bound_
+
+
+def test_gaussian_mixture_setting_best_params():
+    # test that best_params is set appropriately
+    # even if convergence was not reached
+    # also a regression test for issue #18216
```

Comment:
Ideally, we would like to create another test for `BayesianMixture`, but this might be hard (we need to find a set of initial parameters which would cause it to diverge). You can try to see if the parameters you've in-lined cause this problem for `BayesianMixture`. If it doesn't, it's fine not going down the rabbit hole to find ones that do.

```suggestion
def test_gaussian_mixture_params_setting():
    """`GaussianMixture`'s best_parameters, `n_iter_` and `lower_bound_`
    must be set appropriately in the case of divergence.

    Non-regression test for: https://github.com/scikit-learn/scikit-learn/issues/18216
    """
```